信号 $x(t) = \sin(2\pi (f_0t + \frac{1}{2}kt^2))$ を考える（この信号は チャープ信号 (chirp signal) と呼ばれる）．
ただし，$T$ [s] を信号長とし， $k = (f_1 - f_0) / T$ とする．
$f_0 = 100$ [Hz]，$f_1 = 4000$ [Hz]，$T = 5$ [s] として，
$x(t)$ をサンプリング周波数 $F_s = 16000$ Hz でサンプリングし，保存しなさい．

In [1]:
from pathlib import Path
import numpy as np
import soundfile as sf
from IPython.display import Audio, display

output_dir = Path("outputs/audio")
output_dir.mkdir(parents=True, exist_ok=True)

fs = 16000
T = 5.0
f0 = 100.0
f1 = 4000.0
amplitude = 0.5

k = (f1 - f0) / T

t = np.arange(int(fs * T)) / fs
x = amplitude * np.sin(2 * np.pi * (f0 * t + 0.5 * k * t**2))

output_path = output_dir / "06_chirp_100_to_4000hz.wav"
sf.write(output_path, x, fs)

print(f"k = {k:.3f} Hz/s")
print(f"保存先: {output_path}")
print(f"サンプル数: {len(x)}")
print(f"時間長: {len(x) / fs:.2f} s")

display(Audio(x, rate=fs))


k = 780.000 Hz/s
保存先: outputs/audio/06_chirp_100_to_4000hz.wav
サンプル数: 80000
時間長: 5.00 s


$k=(4000-100)/5=780\,\mathrm{Hz/s}$．
このため，瞬時周波数は $100\,\mathrm{Hz}$ から $4000\,\mathrm{Hz}$ まで線形に増加する．
